diff --git a/node_modules/@capacitor/local-notifications/ios/Plugin/LocalNotificationsPlugin.swift b/node_modules/@capacitor/local-notifications/ios/Plugin/LocalNotificationsPlugin.swift
index d2dee24..6f0c032 100644
--- a/node_modules/@capacitor/local-notifications/ios/Plugin/LocalNotificationsPlugin.swift
+++ b/node_modules/@capacitor/local-notifications/ios/Plugin/LocalNotificationsPlugin.swift
@@ -240,6 +240,12 @@ public class LocalNotificationsPlugin: CAPPlugin {
             content.attachments = try makeAttachments(attachments)
         }
 
+        if #available(iOS 15.0, *), let timeSensitive = notification["timeSensitive"] as? Bool {
+            if timeSensitive {
+                content.interruptionLevel = .timeSensitive
+            }
+        }
+
         return content
     }
 
